pkg/engine/masterarmresources.go

```diff
@@ -52,14 +57,18 @@ func createKubernetesMasterResourcesVMAS(cs *api.ContainerService) []interface{}
 		masterNic := createPrivateClusterNetworkInterface(cs)
 		masterResources = append(masterResources, masterNic)
 
-		provisionJumpbox := cs.Properties.OrchestratorProfile.KubernetesConfig.PrivateJumpboxProvision()
+		var provisionJumpbox bool
+
+		if kubernetesConfig != nil {
+			provisionJumpbox = kubernetesConfig.PrivateJumpboxProvision()
+		}
 
 		if provisionJumpbox {
 			jumpboxVM := createJumpboxVirtualMachine(cs)
 			masterResources = append(masterResources, jumpboxVM)
 			jumpboxIsManagedDisks :=
-				cs.Properties.OrchestratorProfile.KubernetesConfig.PrivateJumpboxProvision() &&
-					cs.Properties.OrchestratorProfile.KubernetesConfig.PrivateCluster.JumpboxProfile.StorageProfile == api.ManagedDisks
+				kubernetesConfig.PrivateJumpboxProvision() &&
+					kubernetesConfig.PrivateCluster.JumpboxProfile.StorageProfile == api.ManagedDisks
```

Comment:
do we know that `JumpboxProfile` will never be nil here?

Reply:
The `PrivateJumpboxProvision()` function checks for `JumpoboxProfile`. So we are covered that way.